inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/scatter_update/scatter_nd_update_kernel_ref.cpp

```diff
@@ -0,0 +1,234 @@
+    k.EnableOutputLayout(DataLayout::bfzyx);
+    k.EnableInputLayout(DataLayout::bfwzyx);
+    k.EnableOutputLayout(DataLayout::bfwzyx);
+    k.EnableTensorOffset();
+    k.EnableTensorPitches();
+    k.EnableBatching();
+    k.EnableDifferentTypes();
+    return k;
+}
+
+static inline std::string GetOrderString(std::vector<std::string>& order) {
+    std::string order_str = order[0];
+    for (size_t i = 1; i < order.size(); i++)
+        order_str += ", " + order[i];
+
+    return order_str;
+}
+
+static inline std::vector<std::string> GetDefaultOrder(size_t size) {
+    std::vector<std::string> default_order;
+    if (size <= 4) {
+        default_order = {"b", "f", "y", "x"};
+    } else if (size == 5) {
+        default_order = {"b", "f", "z", "y", "x"};
+    } else if (size == 6) {
+        default_order = {"b", "f", "w", "z", "y", "x"};
+    }
+
+    return default_order;
+}
+
+ScatterNDUpdateKernelRef::DispatchData ScatterNDUpdateKernelRef::SetDefault(const scatter_nd_update_params& params, const optional_params&, bool is_second) const {
+    DispatchData dispatchData;
+
+    if (!is_second) {
+        const auto& scope = params.output;
+
+    switch (params.inputs[0].GetLayout()) {
+    case DataLayout::bfyx:
+            dispatchData.gws = { scope.X().v, scope.Y().v, scope.Feature().v * scope.Batch().v };
+        break;
+
+    case DataLayout::bfzyx:
+            dispatchData.gws = { scope.X().v * scope.Y().v, scope.Z().v, scope.Feature().v * scope.Batch().v };
+        break;
+
+    case DataLayout::bfwzyx:
+            dispatchData.gws = { scope.X().v * scope.Y().v, scope.Z().v * scope.W().v, scope.Feature().v * scope.Batch().v };
+        break;
+    default:
+        assert(0);
+        break;
+    }
+    }
+    else {
+        auto indices_rank = params.indices_rank;
+        const auto& indices = params.inputs[1];
+        auto indices_dims = indices.LogicalDims();
+
+        if (indices_dims.size() > 1)
+        {
+            std::reverse(indices_dims.begin(), indices_dims.end());
+        }
+
+        dispatchData.indicesLastDim = indices_dims[indices_rank - 1];
+        size_t indices_set_size = 1;
+        for (int i = 0; i < (indices_rank - 1); i++)
+        {
+            indices_set_size *= indices_dims[i];
+        }
+        dispatchData.gws = {1, 1, indices_set_size};
+    }
+
+    dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo);
+
+    return dispatchData;
+}
+
+static std::string GetOutputIndex(const scatter_nd_update_params& params) {
+    std::string output_index_str;
+
+    const auto& output = params.output;
+   
+    const size_t B_LEN = output.Feature().v * output.Z().v * output.W().v * output.Y().v * output.X().v;
+    const size_t F_LEN = output.Z().v * output.W().v * output.Y().v * output.X().v;
+    const size_t Z_LEN = output.W().v * output.Y().v * output.X().v;
+    const size_t W_LEN = output.Y().v * output.X().v;
+    const size_t Y_LEN = output.X().v;
+
```

Comment:
why not use existing definition like OUTPUT_FEATURE_PITCH or OUTPUT_Z_PITCH etc?

Reply:
also the location could be obtained by providing "order" to the "OUTPUT_GET_INDEX" definition.. No need to describe calculation